caikit/core/data_model/base.py

```diff
@@ -830,6 +853,14 @@ def fill_proto(self, proto):
                 elif subproto.DESCRIPTOR.full_name == timestamp.TIMESTAMP_PROTOBUF_NAME:
                     timestamp_proto = timestamp.datetime_to_proto(attr)
                     subproto.CopyFrom(timestamp_proto)
+                # check that this is any of the Union of List types
+                elif subproto.DESCRIPTOR.full_name.endswith("Sequence"):
```

Comment:
This maintains backward compatibility with anyone trying to instantiate a DM object with `StrSequence` . 

So for `SampleTaskSamplePrimitiveModuleTrainRequest`, 

```
union_list=caikit.interfaces.common.data_model.StrSequence(values=["str", "sequence"]),
``` 
and 
```
union_list=["str", "sequence"],
```
both work now.
```suggestion
                elif subproto.DESCRIPTOR.full_name.endswith(
                    "Sequence"
                ) and not issubclass(attr.__class__, DataBase):
```

Reply:
The reason we have to do this is because the way unions of lists is handled is different if they are created in a `DataObject` manually vs if they are present as part of a function signature which gets parsed. #409 will fix this once and for all by making this handling common across both scenarios.